packages/components/src/components/Button/Button.tsx

```diff
@@ -62,6 +79,33 @@ const getIcon = (iconName: IconNames, size: ButtonSizeOptions) => {
   return <Icon decorative icon={iconName} size={iconProps(size)} />;
 };
 
+const getTextColor = (
+  variant: ButtonVariantOptions
+): Record<string, string> => {
+  if (variant === "text") {
+    return {
+      active: theme.colors.colorTextLink,
+      hover: theme.colors.colorTextLinkStrong,
+      disabled: theme.colors.colorText,
+    };
+  }
+
+  return {
+    active: theme.colors.colorTextInverse,
+  };
+};
+
+const InnerButton = styled(Box.button)`
```

Comment:
Unfortunately, when we add the `as` prop to a component, styled-components changes the type internally, which has a color property (type string), and create incompatibility with the xstyled prop. This was the workaround I found for it.